src/GuppyDataExplorer/GuppyDataExplorer.jsx

```diff
@@ -13,29 +13,107 @@ import {
   ChartConfigType,
 } from './configTypeDef';
 import './GuppyDataExplorer.css';
+import isEnabled from '../helpers/featureFlags';
 
 class GuppyDataExplorer extends React.Component {
   constructor(props) {
     super(props);
+    let initialState = {};
+    let initialFilter = {};
+
+    if (isEnabled('explorerStoreFilterInURL')) {
+      const stateFromURL = getQueryParameter('filters');
+      if (stateFromURL) {
+        try {
+          const decodedFilter = atob(stateFromURL);
+          const isValidJSON = IsValidJSONString(decodedFilter);
+          if (isValidJSON) {
+            initialState = JSON.parse(decodedFilter);
```

Comment:
is this needed? can you do `JSON.parse(decodedFilter).filter` instead since you already has validated the JSON?